File: backend/app/schemas/workflow.py
from typing import Any

from pydantic import BaseModel, Field


class TaskConfig(BaseModel):
    model: str
    temperature: float = Field(default=0.7, ge=0.0, le=2.0)
    top_p: float | None = Field(default=None, ge=0.0, le=1.0)
    provider_name: str = "ollama"
    structured_output: bool = False


class PromptTemplate(BaseModel):
    system_instruction: str = ""
    user_prompt_template: str


class TaskInput(BaseModel):
    raw_input: str = Field(min_length=1)
    task_type: str
    metadata: dict[str, Any] = Field(default_factory=dict)


class WorkflowRunResult(BaseModel):
    run_id: int
    output_text: str
    output_json: dict[str, Any] | None = None
    prompt_text: str


class PromptExperimentRequest(BaseModel):
    task_input: TaskInput
    task_config: TaskConfig
    prompt_template: PromptTemplate

File: frontend/src/types/workflow.ts
export type TaskInput = {
  raw_input: string
  task_type: string
  metadata?: Record<string, unknown>
}

export type TaskConfig = {
  model: string
  temperature: number
  top_p?: number | null
  provider_name?: string
  structured_output?: boolean
}

export type PromptTemplate = {
  system_instruction: string
  user_prompt_template: string
}

export type PromptExperimentPayload = {
  task_input: TaskInput
  task_config: TaskConfig
  prompt_template: PromptTemplate
}

export type WorkflowRunResult = {
  run_id: number
  output_text: string
  output_json: Record<string, unknown> | null
  prompt_text: string
}

export type RunRecord = {
  id: number
  task_type: string
  input_text: string
  config_json: Record<string, unknown>
  prompt_text: string
  model_name: string
  provider_name: string
  output_text: string
  output_json: Record<string, unknown> | null
  user_preference: string | null
  created_at: string
}
